html/pfappserver/lib/pfappserver/Form/Config/SelfService.pm

```diff
@@ -21,40 +21,62 @@ has_field 'id' =>
    type => 'Text',
    label => 'Profile Name',
    required => 1,
-   messages => { required => 'Please specify a name of the Device Registration entry.' },
-   apply => [ pfappserver::Base::Form::id_validator('device registration ID') ]
+   messages => { required => 'Please specify a name of the Self Service Portal entry.' },
+   apply => [ pfappserver::Base::Form::id_validator('self service ID') ]
   );
 
 has_field 'description' =>
   (
    type => 'Text',
-   messages => { required => 'Please specify the description of the Device Registration entry.' },
+   messages => { required => 'Please specify the description of the Self Service Portal entry.' },
   );
 
-has_field 'category' =>
+has_field 'roles_allowed_to_unregister' =>
   (
    type => 'Select',
-   label => 'Role',
+   label => 'Allowed roles',
+   multiple => 1,
+   element_class => ['chzn-deselect'],
+   element_attr => {'data-placeholder' => 'Click to add a role'},
+   options_method => \&options_roles,
+   tags => { after_element => \&help,
+             help => 'The list of roles that are allowed to unregister devices using the self-service portal. Leaving this empty will allow all users to unregister their devices.' },
+  );
+
+has_field 'device_registration_role' =>
+  (
+   type => 'Select',
+   label => 'Role to assign',
    options_method => \&options_roles,
    tags => { after_element => \&help,
-             help => 'The role to assign to devices registered from the specific portal. If none is specified, the role of the registrant is used.' },
+             help => 'The role to assign to devices registered from the self-service portal. If none is specified, the role of the registrant is used.' },
   );
 
-has_field 'allowed_devices' =>
+has_field 'device_registration_allowed_devices' =>
   (
    type => 'FingerbankSelect',
    multiple => 1,
-   label => 'OS',
+   label => 'Allowed OS',
    element_class => ['chzn-deselect'],
    element_attr => {'data-placeholder' => 'Click to add an OS'},
    tags => { after_element => \&help,
-             help => 'List of OS which will be allowed to be register via the self service portal.' },
+             help => 'List of OS which will be allowed to be registered via the self service portal.' },
    fingerbank_model => "fingerbank::Model::Device",
   );
 
-has_block definition =>
+has_block global_definition =>
```

Comment:
Why change our current naming convention for the main block?

If you really feel a change is needed then to do a search and replace on all the forms and templates.

Reply:
Not sure how that got there, I'll revert the change

Reply:
Actually, scratch my previous comment didn't look at the line.

I actually made it like this because there are multiple blocks which represents different "sections" of the self-service portal

There is one for device registration, one for status page, so the one that applied to both had to be called 'global_definition' to have it differentiate from the rest of the definition in the other forms which represent the whole config definition not sure a part of it

Reply:
Even then our current convention is to have a definition block then additional blocks.
This is just a matter of consistency.

If you want to change the convention I have no issue with that just change it everywhere.

```
html/pfappserver/lib/pfappserver/Form/Config/Domain.pm
html/pfappserver/lib/pfappserver/Form/Config/ProfileCommon.pm
html/pfappserver/lib/pfappserver/Form/Config/Source/Mirapay.pm
html/pfappserver/lib/pfappserver/Form/Config/Provisioning/opswat.pm
html/pfappserver/lib/pfappserver/Form/Config/Provisioning/mobileconfig.pm
html/pfappserver/lib/pfappserver/Form/Config/Switch.pm
html/pfappserver/lib/pfappserver/Form/Config/Source.pm
```